utils/cautils/acmeutils.go

```diff
@@ -402,37 +414,23 @@ func (af *acmeFlow) GetCertificate() ([]*x509.Certificate, error) {
 				Value: af.subject,
 			})
 		}
-		orderPayload, err = json.Marshal(struct {
-			Identifiers []acme.Identifier
-		}{Identifiers: idents})
-		if err != nil {
-			return nil, errors.Wrap(err, "error marshaling new letsencrypt order request")
-		}
+		nor = struct{ Identifiers []acme.Identifier }{Identifiers: idents}
+	}
+
+	if root := af.ctx.String("root"); root == "" {
```

Comment:
I'm confused, because this actually will change the functionality. Before, if you didn't provide a root flag we would still try to load the root from the default location (it didn't need to be in your truststore). But this implementation forces you to either use `--root` or have it in your truststore if you're using a private CA. I'm wary of changing this behavior. Are there options that avoid changing existing behavior? If we are going to change behavior here, I'll need to discuss with the team again.

Reply:
How about
If --root is detected, use that.
If --root is "" but the default root exists, use that.
If not use truststore.

Reply:
The problem that I see here is that if `--root` is not provided, and I do have a root in the default step location, then I won't use the default transport to call a public CA - so this will fail. Which means the behavior will have changed. Let me bring this back up with the team during our next triage.

Reply:
Couldn't you include rather than only use the default step location, thus combining truststore and default step location root cert?

Reply:
That's definitely an option. Need to consider the security implications as well as whether this may cause unexpected behavior for anyone. We've got a triage meeting on wednesdays - will circle back after that.